src/components/KFileUpload/KFileUpload.vue

```diff
@@ -0,0 +1,344 @@
+<template>
+  <div
+    class="k-file-upload w-100"
+  >
+    <KLabel
+      v-if="label && appearance === 'file'"
+      :for="customInputId"
+      v-bind="labelAttributes"
+      data-testid="k-file-upload-label"
+    >
+      {{ label }}
+    </KLabel>
+
+    <KInput
+      :id="customInputId"
+      ref="fileInput"
+      :key="fileInputKey"
+      type="file"
+      :accept="accept"
+      :help="help"
+      :max-file-size="maximumFileSize"
+      :class="[
+        'w-100',
+        'upload-input',
+        'cursor-pointer',
+        { 'image-upload': appearance === 'image' },
+        { 'input-error' : hasUploadError }
+      ]"
+      @change="onFileChange"
+    />
+
+    <slot name="icon">
+      <KIcon
+        v-if="appearance === 'image'"
+        :size="iconSize"
+        :color="iconColor"
+        :icon="icon"
+        class="image-upload-icon"
+        @click="updateFile"
+      />
+    </slot>
+
+    <p
+      v-if="appearance === 'image'"
+      class="image-upload-description"
+      @click="updateFile"
```

Comment:
A paragraph tag cannot have focus by default. Can we utilize an anchor tag `<a>` here instead?

Reply:
@adamdehaven I think we also found that you need to specify `href="#"` to get keyboard nav. Is that right?

Reply:
Yep